### ISSUE TITLE
SQLFluff does not work anymore with dbt_utils.star

### ISSUE THREAD

**Issue by Jrmyy** [sqlfluff/sqlfluff#4861]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This seems to be the same as https://github.com/sqlfluff/sqlfluff/issues/2202 but not entirely related since I am using the dbt templater.

- `dbt_utils.star` [macro](https://github.com/dbt-labs/dbt-utils/blob/1.0.0/macros/sql/star.sql#L17) relies on `flags.WHICH` to be equal to `compile`
- dbt-core sets the `which` flag to compile when using the CLI
- SQLFluff instantiate, in Python, a Compiler but somehow the flag value is `None`, leading to an error for models using `dbt_utils.star`

### Expected Behaviour

The compilation made by SQLFluff DBT Templater should have the `flags.WHICH` set to `compile`

### Observed Behaviour

When models are compiled using SQLFluff DBT templater, the `flags.WHICH` value is `None`

### How to reproduce

* Create a model `my_model.sql`

```sql
SELECT
    {{ dbt_utils.star(
        from=ref('my_other_model'),
        except=['excepted'],
        relation_alias='my_other_model',
        quote_identifiers=False
    ) }}
FROM {{ ref('my_other_model') }}
```

* Run command : `sqlflull lint models/my_model.sql`

The error will be:

> Found unparsable section: 'SELECT\n  /* no columns returned from s...

### Dialect

Athena

### Version

2.0.7

### Configuration

```toml
[tool.sqlfluff]
sql_file_exts = [ ".sql" ]

[tool.sqlfluff.core]
dialect                     = "athena"
templater                   = "dbt"
rules                       = "all,core"
exclude_rules               = "L042,L052,L066,AL05"
max_line_length             = 120
large_file_skip_byte_limit  = 0

[tool.sqlfluff.templater.dbt]
apply_dbt_builtins = true

[tool.sqlfluff.layout.type.comma]
line_position = "leading"

[tool.sqlfluff.indentation]
indent_unit    = "space"
tab_space_size = 4

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"

[tool.sqlfluff.rules.aliasing.table]
aliasing = "explicit"

[tool.sqlfluff.rules.aliasing.column]
aliasing = "explicit"

[tool.sqlfluff.rules.aliasing.expression]
allow_scalar = false

[tool.sqlfluff.rules.capitalisation.identifiers]
extended_capitalisation_policy = "lower"
unquoted_identifiers_policy    = "all"

[tool.sqlfluff.rules.references.consistent]
single_table_references = "consistent"

[tool.sqlfluff.rules.capitalisation.functions]
extended_capitalisation_policy = "lower"

[tool.sqlfluff.rules.aliasing.forbid]
force_enable = false

[tool.sqlfluff.rules.convention.select_trailing_comma]
select_clause_trailing_comma = "forbid"

[tool.sqlfluff.rules.capitalisation.literals]
capitalisation_policy = "lower"

[tool.sqlfluff.rules.structure.subquery]
forbid_subquery_in = "both"

[tool.sqlfluff.rules.ambiguous.column_references]
group_by_and_order_by_style = "explicit"

[tool.sqlfluff.rules.capitalisation.types]
extended_capitalisation_policy = "upper"

[tool.sqlfluff.rules.convention.casting_style]
preferred_type_casting_style = "cast"

[tool.sqlfluff.rules.references.quoting]
prefer_quoted_identifiers = false
prefer_quoted_keywords = true
ignore_words = "level,name,roles,state,uri,users,version"

[tool.sqlfluff.rules.references.keywords]
unquoted_identifiers_policy = "all"
quoted_identifiers_policy = "none"
ignore_words = "level,name,roles,state,uri,users,version"
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by moreaupascal56** [comment]
hey, FYI this is also referenced in this dbt-utils issue: [https://github.com/dbt-labs/dbt-utils/issues/802](https://github.com/dbt-labs/dbt-utils/issues/802)
